format-delimited/src/main/java/io/cdap/plugin/format/delimited/dq/TypeInference.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright © 2017-2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.format.delimited.dq;
+
+import java.math.BigInteger;
+import java.util.regex.Pattern;
+/**
+ * Type Interface provides utility functions that allow you to detect the types of data.
+ */
+public class TypeInference {
+  private static final Pattern patternInteger = Pattern.compile("^(\\+|-)?\\d+$");
+
+  private static final Pattern patternDouble = Pattern.compile(
```

Comment:
Please add unit tests for the functions in this class.

Reply:
Unit tests added